### PR TITLE
Allow XHR interceptor in local development mode

### PR DESCRIPTION
**Pull request by alabiaga**
follow up to addressing https://github.com/ampproject/amphtml/pull/22600#discussion_r291307673

-allow xhr interceptor in local dev mode and not in validation development mode.
-add viewer xhrInterceptor example.

### CODE REVIEW

**Comment by alabiaga** [comment]
cc\ @zhangsu closed the PR with the json handling change and just went the route of doing the json in the response body as a string. Just an FYI, thanks for your review.

**Comment by alabiaga** [comment]
cc\ @fstanis xhrInterceptor example. Will port over same changes to our partner dev code as it'll be cleaner and less noisy of an example.

**Comment by alabiaga** [comment]
@choumx Hey Will when you get a chance. Thanks

File: test/unit/test-xhr.js [hunk 1]
```diff
@@ -751,9 +752,9 @@ describe
           .then(() => expect(sendMessageStub).to.not.have.been.called);
       });
 
-      it('should intercept if viewer untrusted but dev mode', () => {
+      it('should intercept if viewer untrusted but in local dev mode', () => {
         sandbox.stub(viewer, 'isTrustedViewer').returns(Promise.resolve(false));
-        interceptionEnabledWin.AMP_DEV_MODE = true;
+        sandbox.stub(mode, 'getMode').returns({localDev: true});
```
Comment (choumx) [inline test/unit/test-xhr.js]: You can set `test.win.AMP_MODE = {localDev: true};` to avoid the wildcard import.

Comment (alabiaga) [reply test/unit/test-xhr.js]: The test is actually not enclosed in the scenario test cases which mocks the window obj which is specific to testing native vs polyfill fetch, so this wouldn't work. Thanks for the input though.

**Pull request merged**